batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -258,14 +312,47 @@ static void runFhirFetch(FhirEtlOptions options, FhirContext fhirContext) throws
 		p.run().waitUntilFinish();
 	}
 	
-	public static void main(String[] args) throws CannotProvideCoderException {
+	static void runFhirJdbcFetch(FhirEtlOptions options, FhirContext fhirContext) throws PropertyVetoException, IOException {
+		Pipeline p = Pipeline.create(options);
+		JdbcFhirMode jdbcUtil = new JdbcFhirMode();
+		JdbcIO.DataSourceConfiguration jdbcConfig = jdbcUtil.getJdbcConfig(options);
+		int batchSize = options.getBatchSize();
+		ParquetUtil parquetUtil = new ParquetUtil(fhirContext, options.getOutputParquetBase());
+		LinkedHashMap<String, String> reversHashMap = jdbcUtil.createFhirReverseMap(options);
+		// process each table-resource mappings
+		for (Map.Entry<String, String> entry : reversHashMap.entrySet()) {
+			String tableName = entry.getKey();
+			String resourceType = entry.getValue();
+			Schema schema = parquetUtil.getResourceSchema(resourceType);
+			// generate fetching ranges i.e 1 to 1000, 1001 to 2001, depending with batchSize
+			PCollection<KV<String, Iterable<Integer>>> ranges = JdbcFhirMode.createChunkRanges(tableName, batchSize, p,
+			    jdbcConfig);
+			// fetch FHIR resources and generate generic records
+			PCollection<GenericRecord> genericRecords = JdbcFhirMode
+			        .generateFhirUrl(tableName, resourceType, batchSize, ranges, jdbcConfig)
```

Comment:
Suggestion: Try to use `DoFn` and `PTransform` (if you want to bundle several `DoFn`s) and just chain `apply` calls when building the pipeline. That might be a little more readable than passing `PColleciton` to functions.

Reply:
Noted